featurebyte/query_graph/node/function.py

```diff
@@ -0,0 +1,228 @@
+"""
+This module contains generic function related node classes
+"""
+# DO NOT include "from __future__ import annotations" as it will trigger issue for pydantic model nested definition
+from typing import Any, Dict, List, Literal, Optional, Sequence, Tuple, Union
+from typing_extensions import Annotated
+
+from abc import abstractmethod  # pylint: disable=wrong-import-order
+
+from pydantic import BaseModel, Field
+
+from featurebyte.enum import DBVarType, FunctionParameterInputForm
+from featurebyte.models.base import PydanticObjectId
+from featurebyte.query_graph.enum import NodeOutputType, NodeType
+from featurebyte.query_graph.node.base import BaseSeriesOutputNode
+from featurebyte.query_graph.node.metadata.operation import (
+    DerivedDataColumn,
+    NodeOutputCategory,
+    OperationStructure,
+    OperationStructureBranchState,
+    OperationStructureInfo,
+    PostAggregationColumn,
+)
+from featurebyte.query_graph.node.metadata.sdk_code import (
+    CodeGenerationConfig,
+    CodeGenerationContext,
+    ExpressionStr,
+    InfoDict,
+    RightHandSide,
+    StatementT,
+    ValueStr,
+    VariableNameGenerator,
+    VarNameExpressionInfo,
+)
+from featurebyte.query_graph.node.scalar import ValueParameterType
+
+
+class BaseFunctionParameterInput(BaseModel):
+    """BaseFunctionParameterInput class"""
+
+    dtype: DBVarType
+
+    @abstractmethod
+    def get_column_args(self) -> List[Optional[str]]:
```

Comment:
curious - when we would pass in `None` as a column arg?

Reply:
https://github.com/featurebyte/featurebyte/pull/1463#discussion_r1247488091

When we derivate a new column from existing column, the new column's name is `None`.

Reply:
oh ok, just saw that!